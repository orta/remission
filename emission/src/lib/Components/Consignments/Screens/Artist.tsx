import * as React from "react"

import ConsignmentBG from "../Components/ConsignmentBG"
import DoneButton from "../Components/DoneButton"
import ArtistSearch from "../Components/SearchResults"

import { ConsignmentSetup, SearchResult } from "../index"

import { debounce } from "lodash"
import { NavigatorIOS, Route, ScrollView, View, ViewProperties } from "react-native"
import metaphysics from "../../../metaphysics"

interface ArtistSearchResponse {
  match_artist: SearchResult[]
}

interface Props extends ConsignmentSetup, ViewProperties {
  navigator: NavigatorIOS
  route: Route
  updateWithArtist?: (result: SearchResult) => void
}

interface State {
  query: string
  searching: boolean
  results: any | null
}

export default class Artist extends React.Component<Props, State> {
  constructor(props) {
    super(props)
    this.state = {
      query: null,
      searching: false,
      results: null,
    }
  }

  doneTapped = () => {
    this.props.navigator.pop()
  }

  artistSelected = (result: SearchResult) => {
    this.props.updateWithArtist(result)
    this.props.navigator.pop()
  }

  textChanged = async (text: string) => {
    this.setState({ query: text, searching: text.length > 0 })
    this.searchForQuery(text)
  }

  searchForQuery = async (query: string) => {
    const results = await metaphysics<ArtistSearchResponse>(`
      {
        match_artist(term: "${query}") {
          id
          name
          image {
            url
          }
        }
      }
    `)
    this.setState({ results: results.match_artist, searching: false })
  }

  render() {
    return (
      <ConsignmentBG>
        <DoneButton onPress={this.doneTapped}>
          <View
            style={{ alignContent: "center", justifyContent: "flex-end", flexGrow: 1, marginLeft: 20, marginRight: 20 }}
          >
            <ArtistSearch
              results={this.state.results}
              query={this.state.query}
              placeholder="Artist/Designer Name"
              noResultsMessage="Unfortunately we are not accepting consignments for works by"
              onChangeText={this.textChanged}
              searching={this.state.searching}
              resultSelected={this.artistSelected}
            />
          </View>
        </DoneButton>
      </ConsignmentBG>
    )
  }
}
