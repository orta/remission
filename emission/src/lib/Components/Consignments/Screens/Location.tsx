import * as React from "react"

import ConsignmentBG from "../Components/ConsignmentBG"
import DoneButton from "../Components/DoneButton"
import Search from "../Components/SearchResults"

import { ConsignmentSetup, SearchResult } from "../index"

import { debounce } from "lodash"
import { NavigatorIOS, Route, ScrollView, View, ViewProperties } from "react-native"
import metaphysics from "../../../metaphysics"

import { stringify } from "qs"

import { NativeModules } from "react-native"
const { Emission } = NativeModules

interface Props extends ConsignmentSetup, ViewProperties {
  navigator: NavigatorIOS
  route: Route
  updateWithResult?: (city: string, state: string, country: string) => void
}

interface State {
  query: string
  searching: boolean
  results: any | null
}

export default class Location extends React.Component<Props, State> {
  constructor(props) {
    super(props)
    this.state = {
      query: null,
      searching: false,
      results: null,
    }
  }

  doneTapped = () => {
    this.props.navigator.pop()
  }

  locationSelected = async (result: SearchResult) => {
    const apiKey = Emission.googleMapsAPIKey || "AIzaSyBJRIy_zCXQ7XYt9Ubn8bpUIEAxEOKUmx8"
    const queryString = stringify({
      key: apiKey,
      placeid: result.id,
    })

    const response = await fetch("https://maps.googleapis.com/maps/api/place/details/json?" + queryString)
    const results = await response.json()

    const { address_components } = results.result
    const cityPlace = address_components.find(comp => comp.types[0] === "locality")
    const statePlace = address_components.find(comp => comp.types[0] === "administrative_area_level_1")
    const countryPlace = address_components.find(comp => comp.types[0] === "country")

    const city = cityPlace && cityPlace.long_name
    const country = countryPlace && countryPlace.long_name
    const state = statePlace && statePlace.long_name

    this.props.updateWithResult(city, state, country)
    this.props.navigator.pop()
  }

  textChanged = async (text: string) => {
    this.setState({ query: text, searching: text.length > 0 })
    this.searchForQuery(text)
  }

  // https://developers.google.com/places/
  // https://developers.google.com/places/web-service/details
  searchForQuery = async (query: string) => {
    // The 2nd is a throw-away key on a throw-away account
    const apiKey = Emission.googleMapsAPIKey || "AIzaSyBJRIy_zCXQ7XYt9Ubn8bpUIEAxEOKUmx8"
    const queryString = stringify({
      key: apiKey,
      language: "en",
      types: "(cities)",
      input: query,
    })

    const response = await fetch("https://maps.googleapis.com/maps/api/place/autocomplete/json?" + queryString)
    const results = await response.json()

    this.setState({
      results: results.predictions && results.predictions.map(this.predictionToResult),
      searching: false,
    })
  }

  predictionToResult = prediction => ({
    id: prediction.place_id,
    name: prediction.description,
  })

  render() {
    return (
      <ConsignmentBG>
        <DoneButton onPress={this.doneTapped}>
          <View
            style={{ alignContent: "center", justifyContent: "flex-end", flexGrow: 1, marginLeft: 20, marginRight: 20 }}
          >
            <Search
              results={this.state.results}
              query={this.state.query}
              onChangeText={this.textChanged}
              searching={this.state.searching}
              resultSelected={this.locationSelected}
              preImage={require("../../../../../images/consignments/map-pin.png")}
              placeholder="City, Country"
              noResultsMessage="Could not find"
            />
          </View>
        </DoneButton>
      </ConsignmentBG>
    )
  }
}
