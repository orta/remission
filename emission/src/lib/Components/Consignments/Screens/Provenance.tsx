import * as React from "react"

import ConsignmentBG from "../Components/ConsignmentBG"
import DoneButton from "../Components/DoneButton"

import { NavigatorIOS, Route, View, ViewProperties } from "react-native"
import TextArea, { TextAreaProps } from "../Components/TextArea"
import { ConsignmentSetup } from "../index"

interface Props extends ConsignmentSetup, ViewProperties {
  navigator: NavigatorIOS
  route: Route
  updateWithProvenance?: (provenance: string) => void
}

interface State {
  provenance: string
}

export default class Provenance extends React.Component<Props, State> {
  constructor(props) {
    super(props)
    this.state = {
      provenance: null,
    }
  }

  doneTapped = () => {
    this.props.updateWithProvenance(this.state.provenance)
    this.props.navigator.pop()
  }

  textChanged = text => {
    this.setState({ provenance: text })
  }

  render() {
    return (
      <ConsignmentBG>
        <DoneButton onPress={this.doneTapped}>
          <View
            style={{
              alignContent: "center",
              justifyContent: "flex-end",
              flexGrow: 1,
              marginLeft: 20,
              marginRight: 20,
              marginTop: 20,
            }}
          >
            <TextArea
              text={{
                onChangeText: this.textChanged,
                value: this.props.provenance,
                placeholder:
                  "Add notes about how you aquired the work. If you’re not sure add any details about how long you’ve had the work.",
                autoFocus: typeof jest === "undefined" /* TODO: https://github.com/facebook/jest/issues/3707 */,
              }}
            />
          </View>
        </DoneButton>
      </ConsignmentBG>
    )
  }
}
