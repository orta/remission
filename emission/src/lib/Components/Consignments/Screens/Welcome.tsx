import * as React from "react"

import Circle from "../Components/CircleImage"
import ConsignmentBG from "../Components/ConsignmentBG"
import { Button } from "../Components/FormElements"
import { BodyText as P, LargeHeadline } from "../Typography"
import Overview from "./Overview"

import { NavigatorIOS, Route, ScrollView, View, ViewProperties } from "react-native"

interface Props extends ViewProperties {
  navigator: NavigatorIOS
  route: Route
}

export default class Welcome extends React.Component<Props, null> {
  goTapped = () => this.props.navigator.push({ component: Overview })

  render() {
    return (
      <ConsignmentBG>
        <ScrollView style={{ flex: 1 }}>
          <View style={{ paddingTop: 40, alignItems: "center" }}>
            <LargeHeadline>Sell works from your collection through our partner network</LargeHeadline>

            <View style={{ width: 300, alignItems: "center", marginTop: 20 }}>
              <Circle source={require("../../../../../images/consignments/email.png")} />

              <P>Sell work from your collection through our partner network.</P>
              <Circle source={require("../../../../../images/consignments/hammer.png")} />

              <P>Get your work placed in an upcoming sale.</P>
            </View>

            <Button text="GET STARTED" onPress={this.goTapped} />
          </View>
        </ScrollView>
      </ConsignmentBG>
    )
  }
}
