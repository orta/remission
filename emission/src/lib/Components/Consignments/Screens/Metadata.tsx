import * as React from "react"

import {
  LayoutAnimation,
  NavigatorIOS,
  Picker,
  Route,
  ScrollView,
  TouchableHighlight,
  TouchableWithoutFeedback,
  View,
  ViewProperties,
} from "react-native"
import ConsignmentBG from "../Components/ConsignmentBG"
import { BodyText, LargeHeadline, Subtitle } from "../Typography"

import { ConsignmentMetadata, SearchResult } from "../"
import TODO from "../Components/ArtworkConsignmentTodo"
import DoneButton from "../Components/DoneButton"
import { Form, Label, Row } from "../Components/FormElements"
import Text from "../Components/TextInput"
import Toggle from "../Components/Toggle"
import Artist from "./Artist"
import Provenance from "./Provenance"
import SelectFromPhotoLibrary from "./SelectFromPhotoLibrary"
import Welcome from "./Welcome"

interface Props extends ViewProperties {
  navigator: NavigatorIOS
  route: Route // this gets set by NavigatorIOS
  updateWithMetadata?: (result: ConsignmentMetadata) => void
  metadata: ConsignmentMetadata
}

// See: https://github.com/artsy/force/blob/814a03a579290eaac74f910a0db28c2afd9b1753/desktop/apps/consign/client/reducers.js#L22-L38
const categoryOptions = [
  { name: "Painting", value: "PAINTING" },
  { name: "Sculpture", value: "SCULPTURE" },
  { name: "Photography", value: "PHOTOGRAPHY" },
  { name: "Print", value: "PRINT" },
  { name: "Drawing, Collage or other Work on Paper", value: "DRAWING_COLLAGE_OR_OTHER_WORK_ON_PAPER" },
  { name: "Mixed Media", value: "MIXED_MEDIA" },
  { name: "Performance Art", value: "PERFORMANCE_ART" },
  { name: "Installation", value: "INSTALLATION" },
  { name: "Video/Film/Animation", value: "VIDEO_FILM_ANIMATION" },
  { name: "Architecture", value: "ARCHITECTURE" },
  { name: "Fashion Design and Wearable Art", value: "FASHION_DESIGN_AND_WEARABLE_ART" },
  { name: "Jewelry", value: "JEWELRY" },
  { name: "Design/Decorative Art", value: "DESIGN_DECORATIVE_ART" },
  { name: "Textile Arts", value: "TEXTILE_ARTS" },
  { name: "Other", value: "OTHER" },
]

interface State extends ConsignmentMetadata {
  showSelector?: boolean
}

export default class Metadata extends React.Component<Props, State> {
  constructor(props) {
    super(props)
    this.state = props.metadata || {}
  }

  doneTapped = () => {
    this.props.updateWithMetadata(this.state)
    this.props.navigator.pop()
  }

  updateUnit = () => this.setState({ unit: this.state.unit === "CM" ? "IN" : "CM" })
  updateTitle = title => this.setState({ title })
  updateYear = year => this.setState({ year })
  updateCategory = category => this.setState({ category })
  updateMedium = medium => this.setState({ medium })
  updateWidth = width => this.setState({ width })
  updateHeight = height => this.setState({ height })
  updateDepth = depth => this.setState({ depth })

  animateStateChange = newState => {
    const animate = LayoutAnimation.easeInEaseOut as any
    animate()
    this.setState(newState)
  }

  showCategorySelection = () => this.animateStateChange({ showSelector: true })
  hideCategorySelection = () => this.animateStateChange({ showSelector: false })

  render() {
    return (
      <ConsignmentBG>
        <DoneButton onPress={this.doneTapped}>
          <ScrollView keyboardShouldPersistTaps="always">
            <View style={{ padding: 10 }}>
              <Row>
                <Text
                  text={{ placeholder: "Title", onChangeText: this.updateTitle, value: this.state.title }}
                  style={{ margin: 10 }}
                />
              </Row>

              <Row>
                <Text
                  text={{ placeholder: "Year", onChangeText: this.updateYear, value: this.state.year }}
                  style={{ margin: 10 }}
                />
              </Row>

              <Row>
                <TouchableWithoutFeedback onPress={() => console.log("HI")} style={{ flex: 1, height: 40 }}>
                  <Text
                    text={{
                      placeholder: "Category",
                      onChangeText: this.updateCategory,
                      editable: false,
                      value: this.state.category,
                    }}
                    style={{ margin: 10 }}
                  />
                </TouchableWithoutFeedback>
              </Row>

              {this.state.showSelector &&
                <Picker
                  style={{ backgroundColor: "white" }}
                  selectedValue={this.state.category}
                  onValueChange={(itemValue, itemIndex) =>
                    this.setState({ category: itemValue, categoryName: categoryOptions[itemIndex].name })}
                >
                  {categoryOptions.map(category => <Picker.Item label={category.name} value={category.value} />)}
                </Picker>}

              <Row>
                <Text
                  text={{ placeholder: "Medium", onChangeText: this.updateMedium, value: this.state.medium }}
                  style={{ margin: 10 }}
                />
              </Row>

              <Row>
                <Text
                  text={{ placeholder: "Width", onChangeText: this.updateWidth, value: this.state.width }}
                  style={{ margin: 10 }}
                />
                <Text
                  text={{
                    placeholder: "Height",
                    onChangeText: this.updateHeight,
                    value: this.state.height,
                  }}
                  style={{ margin: 10 }}
                />
              </Row>

              <Row>
                <Text text={{ placeholder: "Depth", onChange: this.updateDepth }} style={{ margin: 10 }} />
                <View style={{ flex: 1, flexDirection: "row", alignItems: "center", margin: 10 }}>
                  <Label>Units</Label>
                  <Toggle selected={this.state.unit === "CM"} left="CM" right="IN" onPress={this.updateUnit} />
                </View>
              </Row>
            </View>
          </ScrollView>
        </DoneButton>
      </ConsignmentBG>
    )
  }
}
