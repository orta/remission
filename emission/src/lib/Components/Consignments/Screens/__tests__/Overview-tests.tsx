import * as React from "react"
import "react-native"
import * as renderer from "react-test-renderer"

import Artist from "../Artist"
import Location from "../Location"
import Metadata from "../Metadata"
import Overview from "../Overview"

import Provenance from "../Provenance"
import SelectFromPhotoLibrary from "../SelectFromPhotoLibrary"
import Welcome from "../Welcome"

const nav = {} as any
const route = {} as any

const anything = expect.anything

it("Sets up the right view hierarchy", () => {
  const tree = renderer.create(<Overview navigator={nav} route={route} setup={{}} />).toJSON()
  expect(tree).toMatchSnapshot()
})

describe("Opening the right page", () => {
  let navigator = {} as any
  let overview: Overview

  beforeEach(() => {
    navigator = { push: jest.fn() } as any
    overview = new Overview({ navigator, route, setup: {} })
  })

  it("pushes a Location when you tap on Locations", () => {
    overview.goToLocationTapped()
    expect(navigator.push).toBeCalledWith({ component: Location, passProps: anything() })
  })

  it("pushes a Artist when you tap on Artist", () => {
    overview.goToArtistTapped()
    expect(navigator.push).toBeCalledWith({ component: Artist, passProps: anything() })
  })

  it("pushes a Metadata when you tap on Metadata", () => {
    overview.goToMetadataTapped()
    expect(navigator.push).toBeCalledWith({ component: Metadata, passProps: anything() })
  })

  it("pushes a Provenance when you tap on Provenance", () => {
    overview.goToProvenanceTapped()
    expect(navigator.push).toBeCalledWith({ component: Provenance, passProps: anything() })
  })

  it("pushes a SelectFromPhotoLibrary when you tap on Photos", () => {
    overview.goToPhotosTapped()
    expect(navigator.push).toBeCalledWith({ component: SelectFromPhotoLibrary, passProps: anything() })
  })
})

describe("Updating State", () => {
  let overview: Overview
  let stateMock: jest.Mock<any>
  let update: any

  beforeEach(() => {
    overview = new Overview({ navigator, route, setup: {} })
    update = overview.updateLocalStateAndMetaphysics
    overview.setState = jest.fn()
    stateMock = overview.setState as any
  })

  it("updates Location", () => {
    overview.updateLocation("Huddersfield", "Yorkshire", "UK")
    expect(stateMock).toBeCalledWith({ location: { city: "Huddersfield", country: "UK", state: "Yorkshire" } }, update)
  })

  it("updates Artist", () => {
    overview.updateArtist({ id: "banksy", name: "Banksy" })
    expect(stateMock).toBeCalledWith({ artist: { id: "banksy", name: "Banksy" } }, update)
  })

  it("updates work metadata", () => {
    overview.updateMetadata({
      title: "OK",
      year: "1983",
      category: "PAINTING",
      categoryName: "Painting",
      medium: "Oil on Canvas",
      width: "100",
      height: "200",
      depth: 20,
      unit: "cm",
      displayString: "Paint on canvas",
    })

    expect(stateMock).toBeCalledWith(
      {
        metadata: {
          category: "PAINTING",
          categoryName: "Painting",
          depth: 20,
          displayString: "Paint on canvas",
          height: "200",
          medium: "Oil on Canvas",
          title: "OK",
          unit: "cm",
          width: "100",
          year: "1983",
        },
      },
      update
    )
  })

  it("updates Provenance", () => {
    overview.updateProvenance("This is my provenance")
    expect(stateMock).toBeCalledWith({ provenance: "This is my provenance" }, update)
  })

  it.skip("updates Photos")
})
