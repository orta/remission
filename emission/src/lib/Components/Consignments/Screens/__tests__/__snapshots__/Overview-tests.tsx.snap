// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sets up the right view hierarchy 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "black",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "height": "100%",
          "maxWidth": 540,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "paddingTop": 18,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 30,
                  "marginBottom": 20,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "textAlign": "center",
                },
                Object {
                  "marginLeft": 40,
                  "marginRight": 40,
                },
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }
          >
            Complete work details to submit
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#cccccc",
                  "fontSize": 20,
                  "paddingLeft": 25,
                  "paddingRight": 25,
                },
                Object {
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "AGaramondPro-Italic",
                },
              ]
            }
          >
            Provide as much detail as possible so that our partners can best assess your work.
          </Text>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "black",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "paddingBottom": 20,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 20,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#e5e5e5",
                    "height": 1,
                  },
                  undefined,
                ]
              }
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 60,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderColor": "white",
                        "borderRadius": 13,
                        "borderWidth": 1,
                        "height": 26,
                        "justifyContent": "center",
                        "marginRight": 20,
                        "width": 26,
                      },
                      undefined,
                    ]
                  }
                >
                  <Image
                    source={1}
                  />
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "Avant Garde Gothic ITCW01Dm",
                      },
                      undefined,
                    ]
                  }
                >
                  ARTIST/DESIGNER
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "AGaramondPro-Regular",
                        "paddingTop": 6,
                        "textAlign": "right",
                      },
                      undefined,
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#e5e5e5",
                    "height": 1,
                  },
                  undefined,
                ]
              }
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 60,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderColor": "white",
                        "borderRadius": 13,
                        "borderWidth": 1,
                        "height": 26,
                        "justifyContent": "center",
                        "marginRight": 20,
                        "width": 26,
                      },
                      undefined,
                    ]
                  }
                >
                  <Image
                    source={1}
                  />
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "Avant Garde Gothic ITCW01Dm",
                      },
                      undefined,
                    ]
                  }
                >
                  PHOTOS
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#e5e5e5",
                    "height": 1,
                  },
                  undefined,
                ]
              }
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 60,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderColor": "white",
                        "borderRadius": 13,
                        "borderWidth": 1,
                        "height": 26,
                        "justifyContent": "center",
                        "marginRight": 20,
                        "width": 26,
                      },
                      undefined,
                    ]
                  }
                >
                  <Image
                    source={1}
                  />
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "Avant Garde Gothic ITCW01Dm",
                      },
                      undefined,
                    ]
                  }
                >
                  METADATA
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "AGaramondPro-Regular",
                        "paddingTop": 6,
                        "textAlign": "right",
                      },
                      undefined,
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#e5e5e5",
                    "height": 1,
                  },
                  undefined,
                ]
              }
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 60,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderColor": "white",
                        "borderRadius": 13,
                        "borderWidth": 1,
                        "height": 26,
                        "justifyContent": "center",
                        "marginRight": 20,
                        "width": 26,
                      },
                      undefined,
                    ]
                  }
                >
                  <Image
                    source={1}
                  />
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "Avant Garde Gothic ITCW01Dm",
                      },
                      undefined,
                    ]
                  }
                >
                  LOCATION
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "AGaramondPro-Regular",
                        "paddingTop": 6,
                        "textAlign": "right",
                      },
                      undefined,
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#e5e5e5",
                    "height": 1,
                  },
                  undefined,
                ]
              }
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 60,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderColor": "white",
                        "borderRadius": 13,
                        "borderWidth": 1,
                        "height": 26,
                        "justifyContent": "center",
                        "marginRight": 20,
                        "width": 26,
                      },
                      undefined,
                    ]
                  }
                >
                  <Image
                    source={1}
                  />
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "Avant Garde Gothic ITCW01Dm",
                      },
                      undefined,
                    ]
                  }
                >
                  PROVENANCE
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "AGaramondPro-Regular",
                        "paddingTop": 6,
                        "textAlign": "right",
                      },
                      undefined,
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#e5e5e5",
                    "height": 1,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "paddingVertical": 6,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "height": 43,
                  "marginTop": 20,
                  "width": 174,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgba(255, 255, 255, 1)",
                      "borderColor": "#e5e5e5",
                      "borderWidth": 1,
                      "flex": 1,
                      "justifyContent": "center",
                    },
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontSize": 12,
                        },
                        Object {
                          "color": "black",
                          "fontFamily": "AGaramondPro-Regular",
                          "opacity": 1,
                        },
                        Object {
                          "fontFamily": "Avant Garde Gothic ITCW01Dm",
                        },
                      ]
                    }
                  >
                    NEXT
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "position": "absolute",
            "right": 20,
            "top": 20,
            "width": 40,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={1}
      />
    </View>
  </View>
</View>
`;
